pkg/api/2018-09-30-preview/api/types_test.go

```diff
@@ -177,14 +173,21 @@ func TestUnmarshal(t *testing.T) {
 func TestStructTypes(t *testing.T) {
 	// AgentPoolProfile and MasterPoolProfile types should be identical bar
 	// `Role AgentPoolProfileRole` in the former
+	// MasterPoolProfile has removed OsType, Name
 	app := reflect.TypeOf(AgentPoolProfile{})
 	mpp := reflect.TypeOf(MasterPoolProfile{})
-	if app.NumField() != mpp.NumField()+1 {
+	// Add 3 for Role,OsType,Name which are missing from MasterPoolProfile
+	if app.NumField() != mpp.NumField()+3 {
```

Comment:
@jim-minter, I'm not sure this check is useful any more.  The previous one was adding 1 to make up for the Role.  I have added 2 more for OsType and Name.  At this point maybe we just remove it?

Reply:
we can check the diff - please open a follow-up issue